### ISSUE TITLE
Add `anyhastags(::ITensor, ::String)` and `allhastags(::ITensor, ::String)`

### ISSUE THREAD

**Issue by mtfishman** [ITensor/ITensors.jl#491]
It can be useful to check if any or all indices of an ITensor has certain tags (or prime levels, or other things). We could define:
```julia
anyhastags(A::ITensor, ts::String) = any(i -> hastags(i, ts), inds(A))
allhastags(A::ITensor, ts::String) = all(i -> hastags(i, ts), inds(A))
```
Similarly, we could define the same things for IndexSets, MPS and MPO. Also, we could define:
```julia
hastags(A::ITensor, ts::String) = anyhastags(A::ITensors, ts::String)
```
though I'm not sure if it should default to `anyhastags` or `allhastags`.

**Comment by emstoudenmire** [comment]
These definitely sound useful. The ITensor case (last one) is interesing. My instinct would be that it would mean that any of the indices has the tags.

**Comment by mtfishman** [comment]
Closed by #495.
